common/options/options.go

```diff
@@ -372,7 +372,7 @@ type URISetter interface {
 
 func (auth *Auth) RequiresExternalDB() bool {
 	return auth.Mechanism == "GSSAPI" || auth.Mechanism == "PLAIN" ||
-		auth.Mechanism == "MONGODB-X509"
+		auth.Mechanism == "MONGODB-X509" || auth.Mechanism == "MONGODB-OIDC"
```

Comment:
Suggestion: rewrite this list of mechanisms as []slice and check `slices.Contains`.